horace_core/sqw/PixelData/@PixelData/get_pixels.m

```diff
@@ -30,19 +30,34 @@
 abs_pix_indices = parse_args(obj, abs_pix_indices);
 
 if obj.is_file_backed_()
-    first_required_page = ceil(min(abs_pix_indices)/obj.base_page_size);
-    obj.move_to_page(first_required_page);
+    if any(obj.page_dirty_)
+        % At least some pixels sit in temporary files
 
-    pix_out = PixelData(numel(abs_pix_indices));
+        % Allocate output array
+        pix_out = PixelData(numel(abs_pix_indices));
 
-    [pg_idxs, global_idxs] = get_idxs_in_current_page_(obj, abs_pix_indices);
-    pix_out.data(:, global_idxs) = obj.data(:, pg_idxs);
-    while obj.has_more()
-        obj.advance();
-        [pg_idxs, global_idxs] = get_idxs_in_current_page_(obj, abs_pix_indices);
-        pix_out.data(:, global_idxs) = obj.data(:, pg_idxs);
+        % Deal with dirty pixels
+        [pix_out, dirty_pg_mask] = ...
+            assign_dirty_pixels(obj, pix_out, abs_pix_indices);
+
+        % Now assign the clean pixels
+        [unique_sorted, ~, idx_map] = unique(abs_pix_indices(~dirty_pg_mask));
+        raw_pix = obj.f_accessor_.get_pix_at_indices(unique_sorted);
+        pix_out.data(:, ~dirty_pg_mask) = raw_pix(:, idx_map);
+
+    else
+        % All pixels in file
+        if issorted(abs_pix_indices, 'strictascend')
```

Comment:
Is it worth adding a comment noting that the sort/unsort code is significantly slower, even if the data is sorted? That explains why we are doing this check.